src/main/java/org/shanerx/tradeshop/utils/configuration/YAMLConfigurationController.java

```diff
@@ -0,0 +1,248 @@
+import java.io.Writer;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Set;
+import java.util.logging.Level;
+
+public class YAMLConfigurationController {
+
+    protected final TradeShop plugin;
+    protected final File file;
+    protected FileConfiguration config;
+
+    public YAMLConfigurationController(TradeShop plugin, String fileNameNoExtension) {
+        file = new File(plugin.getDataFolder(), fileNameNoExtension + ".yml");
+        config = YamlConfiguration.loadConfiguration(file);
+        this.plugin = plugin;
+    }
+
+    public void addKey(String path, Object value) {
+        if (config.get(path) == null) {
+            config.set(path, value);
+        }
+    }
+
+    public void load() {
+        try {
+            if (!plugin.getDataFolder().isDirectory()) {
+                plugin.getDataFolder().mkdirs();
+            }
+            if (!file.exists()) {
+                file.createNewFile();
+            }
+        } catch (IOException e) {
+            plugin.getLogger().log(Level.SEVERE, "Could not create " + file.getName() + " file! Disabling plugin!", e);
+            plugin.getServer().getPluginManager().disablePlugin(plugin);
+        }
+
+        config = YamlConfiguration.loadConfiguration(file);
+    }
+
+    public void save() {
+        Validate.notNull(file, "File cannot be null");
+
+        if (config != null)
+            try {
+                Files.createParentDirs(file);
+
+                // Map<Section.getPath, outputTest++>
+                HashMap<String, String> outputMap = new HashMap<>();
+
+                StringBuilder data = new StringBuilder();
+
+                String bar = "", name = config.getString("file-name");
+
+                for (int i = name.length() + 10; i > 0; i--) {
+                    bar += "#";
+                }
+
+                data.append(bar).append("\n");
+                data.append("#    ").append(name).append("    #\n");
+                data.append(bar).append("\n");
+
+                Set<YAMLConfigurationSectionInterface> sectionKeys = null;
+                Set<YAMLConfigurationInterface> keys = null;
+
+                if (name.contains("Config")) {
+                    sectionKeys = Sets.newHashSet(SettingSectionKey.values());
+                    keys = Sets.newHashSet(SettingsEnum.values());
+                } else if (name.contains("Messages")) {
+                    sectionKeys = Sets.newHashSet(MessageSectionKey.values());
+                    keys = Sets.newHashSet(MessagesEnum.values());
+                }
+
+                // Create Header for each Section and add it to the map
+                for (YAMLConfigurationSectionInterface section : sectionKeys) {
+                    StringBuilder sectionHeader = new StringBuilder();
+
+                    // If Section has a key add section to file
+                    if (!section.getPath().isEmpty()) {
```

Comment:
maybe de-escalate this by one level with `if (section.getPath().isEmpty()) continue;`

Reply:
I could, but i don't think it would really matter, also this was cleaned up in the current implementation